lib/models/persistent_tab_controller.dart

```diff
@@ -18,7 +39,23 @@ class PersistentTabController extends ChangeNotifier {
       return;
     }
     if (!isUndo) {
-      _tabHistory.add(_index);
+      if (_clearHistoryOnInitialIndex && value == _initialIndex) {
+        _tabHistory.clear();
+      } else {
+        if (_historyLength == 1 && _tabHistory.length == 1 && _tabHistory[0] == value) {
```

Comment:
It might be slimmer but it would be harder to read I guess. Right now it seems clearer (to me at least) what it does. If you really want it, I can merge the conditions, just let me know.